src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -589,6 +589,7 @@ public override Assembly GetSatelliteAssembly(CultureInfo culture, Version? vers
             // This stack crawl mark is never used because the requesting assembly is explicitly specified,
             // so the value could be anything.
             StackCrawlMark unused = default;
+            _ = AssemblyLoadContext.Default; // Ensure default ALC is initialized - the runtime may use it as part of satellite assembly resolution
```

Comment:
Where does the runtime use it? What prevents it being initialized on demand - right when the runtime uses it?

Reply:
Mm, I could make `CLRPrivBinderCoreCLR` call up into some managed function that initializes the managed default ALC if it doesn't already have the managed ALC during an assembly bind. It tries to use the managed ALC for resolution if the assembly wasn't already loaded or in the TPA list: https://github.com/dotnet/runtime/blob/master/src/coreclr/src/binder/clrprivbindercoreclr.cpp#L79-L82

Reply:
Yes, I think something like that would be better.